internal/provider/resource_edge_cluster.go

```diff
@@ -183,22 +181,38 @@ func resourceNsxEdgeClusterCreate(ctx context.Context, data *schema.ResourceData
 		return diag.FromErr(err)
 	}
 
-	clusters, err := client.GetEdgeClustersWithResponse(ctx, nil)
+	const maxRetries = 30
+	const retryDelay = 10 * time.Second
 
-	if err != nil {
-		return diag.FromErr(err)
-	}
-	page, vcfErr := api_client.GetResponseAs[vcf.PageOfEdgeCluster](clusters)
-	if vcfErr != nil {
-		api_client.LogError(vcfErr, ctx)
-		return diag.FromErr(errors.New(*vcfErr.Message))
-	}
+	for attempt := 0; attempt < maxRetries; attempt++ {
+		if attempt > 0 {
+			tflog.Info(ctx, fmt.Sprintf("Retry attempt %d/%d to find edge cluster", attempt, maxRetries))
+		}
```

Comment:
`attempt` will always be >= 0

```suggestion
		tflog.Info(ctx, fmt.Sprintf("Retry attempt %d/%d to find edge cluster", attempt, maxRetries))
```